lib/commons/dom/is-focusable.js

```diff
@@ -65,9 +65,11 @@ dom.isNativelyFocusable = function(el) {
 			return el.type !== 'hidden';
 		case 'TEXTAREA':
 		case 'SELECT':
-		case 'DETAILS':
+		case 'SUMMARY':
 		case 'BUTTON':
 			return true;
+		case 'DETAILS':
+			return !el.querySelector('summary');
```

Comment:
This should test the flattened tree instead. details > summary works across shadow tree boundaries.

Reply:
How does details > summary work across shadow boundaries? Details is not one of the elements you can [attach shadow](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow) to, so you would have to do it to a child of the details. That doesn't property create a summary for the details to use.

```html
<details>
  <div id="shadow"></div>
</details>

<script>
  let details = document.getElementById('shadow');
  let shadow = details.attachShadow({mode: 'open'});
  shadow.innerHTML = `<summary>A summary</summary>
  Some content`;
</script>
```